.github/workflows/docker.yml

```diff
@@ -0,0 +1,64 @@
+name: Docker Release
+
+on:
+  # Trigger the workflow on the new 'v*' tag created
+  push:
+    tags:
+      - "v*"
+
+jobs:
+  docker:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        distro: ["scratch", "alpine", "debian"]
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v1
+        with:
+          platforms: linux/amd64
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+
+      - name: Login to GitHub Container Registry
+        uses: docker/login-action@v1
+        with:
+          registry: ghcr.io
+          username: ${{ github.repository_owner }}
+          password: ${{ secrets.GHCR_IO_REGISTRY_TOKEN }}
+
+      - name: Set target and tag
+        id: build-opts
+        run: |
+          if [ "${{ matrix.distro }}" = 'scratch' ]; then
+            echo ::set-output name=target::''
+            echo ::set-output name=latest::latest
+            echo ::set-output name=tag::"${GITHUB_REF/refs\/tags\/v/}"
+          else
+            echo ::set-output name=target::"${{ matrix.distro }}-distro"
+            echo ::set-output name=latest::"latest-${{ matrix.distro }}"
+            echo ::set-output name=tag::"${GITHUB_REF/refs\/tags\/v/}-${{ matrix.distro }}"
+          fi
```

Comment:
This is where the workflow will set the target and tag for `docker build` command appropriately.